pkg/checksums/checksums.go

```diff
@@ -369,20 +369,36 @@ func (e *Embedder) interpolateTemplate(template string, additionalVars map[strin
 
 // createChecksumFilename creates the checksum filename using the template from the spec
 func (e *Embedder) createChecksumFilename() string {
+	return e.createChecksumFilenameWithAsset("")
+}
+
+// createChecksumFilenameWithAsset creates the checksum filename with optional asset filename support
+func (e *Embedder) createChecksumFilenameWithAsset(assetFilename string) string {
 	if e.Spec.Checksums == nil || spec.StringValue(e.Spec.Checksums.Template) == "" {
 		return ""
 	}
 
 	template := spec.StringValue(e.Spec.Checksums.Template)
 
-	// Check for unsupported ASSET_FILENAME variable
+	// Check for unsupported ASSET_FILENAME variable in embed-checksums mode
 	if strings.Contains(template, "${ASSET_FILENAME}") {
-		log.Errorf("${ASSET_FILENAME} is not supported in checksum templates. Use 'binst embed-checksums --mode calculate' instead.")
-		return ""
+		if e.Mode != "" {
+			log.Errorf("${ASSET_FILENAME} is not supported in checksum templates for embed-checksums. Use 'binst embed-checksums --mode calculate' instead.")
+		}
+		// Return empty string when ASSET_FILENAME is used but no asset filename is provided
+		if assetFilename == "" {
+			return ""
+		}
+	}
+
+	// Build additional variables map
+	additionalVars := make(map[string]string)
+	if assetFilename != "" && strings.Contains(template, "${ASSET_FILENAME}") {
```

Comment:
No need to check `strings.Contains`.
It's actually unreliable since template could include `$ASSET_FILENAME` and it's needless in the first place.